include/sound/sof/topology.h

```diff
@@ -85,7 +86,7 @@ struct sof_ipc_comp_config {
 	struct sof_ipc_cmd_hdr hdr;
 	uint32_t periods_sink;	/**< 0 means variable */
 	uint32_t periods_source;/**< 0 means variable */
-	uint32_t reserved1;	/**< reserved */
+	uint32_t ref_comp_id;	/**< ID of the referenced component */
```

Comment:
@lgirdwood to let the host know where to attach the guest pipeline, to which mixer / demux.

Reply:
@plbossart but if the "base" topology on the host has several connector points for possibly multiple different VMs, how would it know where to attach which guest?
What we have here is a topology on the host side like this:
![tplg](https://user-images.githubusercontent.com/1363683/70811000-14073180-1dc5-11ea-80c9-24d434a3828f.png)
where VIO2.0 and VIO3.0 are the two virtual PCMs that serve as plugin locations for the VM. As long as we only support 1 VM with exactly 1 playback and 1 capture we don't need to exchange components IDs. But if we want to be able to, say, provide 2 playback plugin points for the single VM, we have to distinguish between them.

Reply:
the sub-topology seen by the guest would be VIO -> BUF -> VSINK in that case, it would not see the entire topology.

the serviceOS would absolutely need to know what guest uses what connection, e.g. for critical sounds there cannot be any ambiguity on where sound gets rendered

Reply:
@plbossart almost. The guest sees nothing of that. The guest only gets its topology file, whatever it contains. Additionally it obtains a "reference ID" - a plugin point, the component ID of the mixer / demux, so that the guest then just replaces the sink / source ID of its last buffer with that plugin ID, which then can be sent to the DSP.